test/unit/test_discovery_v2.py

```diff
@@ -0,0 +1,1214 @@
+        send_request(self, body, response)
+        assert len(responses.calls) == 1
+
+    #--------------------------------------------------------
+    # Test 3: Send empty data and check response
+    #--------------------------------------------------------
+    @responses.activate
+    def test_update_training_query_empty(self):
+        check_empty_required_params(self, fake_response_TrainingQuery_json)
+        check_missing_required_params(self)
+        assert len(responses.calls) == 0
+
+    #-----------
+    #- Helpers -
+    #-----------
+    def make_url(self, body):
+        endpoint = '/v2/projects/{0}/training_data/queries/{1}'.format(
+            body['project_id'], body['query_id'])
+        url = '{0}{1}'.format(base_url, endpoint)
+        return url
+
+    def add_mock_response(self, url, response):
+        responses.add(responses.POST,
+                      url,
+                      body=json.dumps(response),
+                      status=201,
+                      content_type='application/json')
+
+    def call_service(self, body):
+        service = DiscoveryV2(authenticator=NoAuthAuthenticator(),
+                              version='2019-11-22')
+        service.set_service_url(base_url)
+        output = service.update_training_query(**body)
+        return output
+
+    def construct_full_body(self):
+        body = dict()
+        body['project_id'] = "string1"
+        body['query_id'] = "string1"
+        body.update({
+            "natural_language_query": "string1",
+            "examples": [],
+            "filter": "string1",
+        })
+        return body
+
+    def construct_required_body(self):
+        body = dict()
+        body['project_id'] = "string1"
+        body['query_id'] = "string1"
+        body.update({
+            "natural_language_query": "string1",
+            "examples": [],
+            "filter": "string1",
+        })
+        return body
+
+
+# endregion
+##############################################################################
+# End of Service: TrainingData
+##############################################################################
+
+
+def check_empty_required_params(obj, response):
+    """Test function to assert that the operation will throw an error when given empty required data
+
+    Args:
+        obj: The generated test function
+
+    """
+    body = obj.construct_full_body()
+    body = {k: None for k in body.keys()}
+    error = False
+    try:
+        send_request(obj, body, response)
+    except ValueError as e:
+        error = True
+    assert error
+
+
+def check_missing_required_params(obj):
+    """Test function to assert that the operation will throw an error when missing required data
+
+    Args:
+        obj: The generated test function
+
+    """
+    body = obj.construct_full_body()
+    url = obj.make_url(body)
+    error = False
+    try:
+        send_request(obj, {}, {}, url=url)
+    except TypeError as e:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [local variable 'e' is assigned to but never used](https://app.codacy.com/app/watson-developer-cloud/python-sdk/pullRequest?prid=4546453)